desktop/core/src/desktop/lib/botserver/views.py

```diff
@@ -81,6 +82,91 @@ def parse_events(request, event):
   if event.get('type') == 'link_shared':
     handle_on_link_shared(request.get_host(), channel_id, event.get('message_ts'), event.get('links'), user_id)
 
+  if event.get('type') == 'app_mention':
+    handle_on_app_mention(request.get_host(), request.is_secure(), channel_id, user_id, event.get('text'))
+
+
+def handle_on_app_mention(host_domain, is_http_secure, channel_id, user_id, text):
+  if text and 'table' in text:
+    user = get_user(channel_id, check_slack_user_permission(host_domain, user_id))
+
+    table_name = text.split('table ', 1)[1]
+    if '.' in table_name:
+      database, table = table_name.split('.')
+      search_table(host_domain, is_http_secure, channel_id, user, table, database)
+    else:
+      search_table(host_domain, is_http_secure, channel_id, user, table_name)
+
+
+def _get_dialects(user):
+  dialect_types = []
+
+  interpreters = get_ordered_interpreters(user)
+  for interpreter in interpreters:
+    if interpreter['is_sql'] or interpreter['interface'] == 'hms':
+      dialect_types.append(interpreter['type'])
+
+  return dialect_types
+
+
+def search_table(host_domain, is_http_secure, channel_id, user, table_name, db_name=None):
+  mock_request = MockRequest(user=rewrite_user(user))
+  dialect_types = _get_dialects(user)
+
+  table_link = []
+  max_links_count = 3
+  for dialect in dialect_types:
+    snippet = { 'type': dialect }
+    try:
+      databases = get_api(mock_request, snippet).autocomplete(snippet)
+    except Exception as e:
+      continue
+
+    for db in databases.get('databases', []):
+      if db_name and db_name != db:
+        continue
+      try:
+        tables = get_api(mock_request, snippet).autocomplete(snippet, database=db)
+      except Exception as e:
+        continue
+
+      for table in tables.get('tables_meta', []):
+        if table['name'] == table_name:
+          if max_links_count == 0:
+            break
+
+          link = _make_table_link(host_domain, is_http_secure, db, table_name, dialect)
+          table_link.append({
+            'url': link,
+            'database': db,
+            'table': table_name,
+            'dialect': dialect,
+          })
+          max_links_count -= 1
```

Comment:
This will never work in real life (100s of DBs)

We should leverage the table search only by reusing the catalog search (cf. Atlas links shown before)